src/OrchardCore/OrchardCore.DisplayManagement/TagHelpers/ZoneTagHelper.cs

```diff
@@ -41,6 +44,14 @@ public async override Task ProcessAsync(TagHelperContext context, TagHelperOutpu
             {
                 await shape.AddAsync(childContent, Position);
             }
+            else
+            {
+                _logger.LogWarning(
+                    "Unable to add shape to the zone using the <zone> tag helper because the zone's type is " +
+                    "\"{ActualType}\" instead of the expected {ExpectedType}",
+                    zone.GetType().FullName,
+                    nameof(Shape));
```

Comment:
`{ExpectedType}` is always supposed to be `"Shape"`. I only made it a parameter because I think it's clearer this way. Do you want me to remove the parameter and just type "Shape" in the message string instead?

Reply:
> {ExpectedType} is always supposed to be "Shape"

No need to remove it, it's fine to keep it this way